core/src/main/java/com/linecorp/armeria/server/HttpHeaderPathMapping.java

```diff
@@ -57,6 +58,7 @@
                           List<MediaType> consumeTypes, List<MediaType> produceTypes) {
         this.pathStringMapping = requireNonNull(pathStringMapping, "pathStringMapping");
         this.supportedMethods = requireNonNull(supportedMethods, "supportedMethods");
+        checkArgument(!supportedMethods.isEmpty(), "supportedMethods can't be empty");
         this.consumeTypes = requireNonNull(consumeTypes, "consumeTypes");
         this.produceTypes = requireNonNull(produceTypes, "produceTypes");
```

Comment:
How about making sure `supportedMethods`, `consumeTypes` and `produceTypes` are all immutable, like using `ImmutableList.copyOf()` or `ImmutableSet.copyOf()`? I guess it will be harder to make sure immutability otherwise.